File: Haskell/Implementations/Decorator/FontColor.hs
module FontColor
    ( FontColor(..)
    , defaultColor
    ) where

import BaseColor (BaseColor(..))
import Color (Color(..))

newtype FontColor = MkFontColor BaseColor

instance Color FontColor where
    colorCode (MkFontColor c) = "\x001b[" <> cc <> "m"
      where
        cc = case c of
            Black     -> "30"
            Red       -> "31"
            Green     -> "32"
            Yellow    -> "33"
            Blue      -> "34"
            Purple    -> "35"
            LightBlue -> "36"
            White     -> "37"

defaultColor :: FontColor
defaultColor = MkFontColor White

File: Java/Implementations/Decorator/FontColor.java
public class FontColor implements Color {
	private BaseColor color;

	public FontColor(BaseColor color) {
		this.color = color;
	}

	private String cc() {
		switch (this.color) {
			case BLACK:     return "30";
			case RED:       return "31";
			case GREEN:     return "32";
			case YELLOW:    return "33";
			case BLUE:      return "34";
			case PURPLE:    return "35";
			case LIGHTBLUE: return "36";
			case WHITE:     return "37";
			default:   return "30";
		}
	}

	public String colorCode() {
		return "\u001b[" + this.cc() + "m";
	}

	public static Color DEFAULT = new FontColor(BaseColor.WHITE);
}
